### ISSUE TITLE
VS is stuck when mouseover "open"

### ISSUE THREAD

**Issue by starship863** [Microsoft/PTVS#4040]
When editing python and move mouse over the function 'open', the VS is stuck immediatly.

VS version: 15.7.0 preview 3
Python version: anaconda 5.1.0
Windows version: 1803 (17133.73)

How to reproduce:
1. create some python file
2. input some python code as: "open(input_file_name, 'r')"
3. move mouse cursor over the function name 'open' above
4. VS is stuck

This problem does not appear when mouse over other functions or variables.

**Comment by huguesv** [comment]
Confirmed on public preview 3. Thanks for reporting this! I'm checking on preview 4 to see if it's already been fixed.

**Comment by huguesv** [comment]
Haven't seen it on preview 4, but I'm not ready to rule it out yet. I have trouble seeing it again on preview 3.

FWIW here's the call stack for main thread, if I break into debugger when it was stuck. I did break a few times, it was always within PerformLayout.
```
>	mscorlib.dll!System.Collections.Generic.List<Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.FormattedLineGroup>.Capacity.set(int value)	Unknown
 	mscorlib.dll!System.Collections.Generic.List<System.__Canon>.EnsureCapacity(int min)	Unknown
 	mscorlib.dll!System.Collections.Generic.List<Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.FormattedLineGroup>.Add(Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.FormattedLineGroup item)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.DoLayoutDown(Microsoft.VisualStudio.Text.ITextSnapshotLine visualLine, double referenceLine, double effectiveViewportHeight, ref double distanceBelowReferenceLine, System.Threading.CancellationToken? cancel)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.DoCompleteLayout(Microsoft.VisualStudio.Text.SnapshotPoint anchorPosition, double verticalDistance, Microsoft.VisualStudio.Text.Editor.ViewRelativePosition relativeTo, double effectiveViewportHeight, out double referenceLine, out double distanceAboveReferenceLine, out double distanceBelowReferenceLine, System.Threading.CancellationToken? cancel)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.InnerPerformLayout(Microsoft.VisualStudio.Text.SnapshotPoint anchorPosition, double verticalDistance, Microsoft.VisualStudio.Text.Editor.ViewRelativePosition relativeTo, double effectiveViewportWidth, double effectiveViewportHeight, bool preserveViewportTop, System.Threading.CancellationToken? cancel)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.PerformLayout(Microsoft.VisualStudio.Text.ITextSnapshot newSnapshot, Microsoft.VisualStudio.Text.ITextSnapshot newVisualSnapshot, Microsoft.VisualStudio.Text.SnapshotPoint anchorPosition, double verticalDistance, Microsoft.VisualStudio.Text.Editor.ViewRelativePosition relativeTo, double effectiveViewportWidth, double effectiveViewportHeight, bool preserveViewportTop, System.Threading.CancellationToken? cancel)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.DisplayTextLineContainingBufferPosition(Microsoft.VisualStudio.Text.SnapshotPoint bufferPosition, double verticalDistance, Microsoft.VisualStudio.Text.Editor.ViewRelativePosition relativeTo, double? viewportWidthOverride, double? viewportHeightOverride)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.AdornmentLibrary.ToolTip.Implementation.WpfObjectViewElementFactory.AutoSizeTextView(Microsoft.VisualStudio.Text.Editor.IWpfTextView textView, double maxWidth, double maxHeight)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.AdornmentLibrary.ToolTip.Implementation.WpfObjectViewElementFactory.CreateViewElement.AnonymousMethod__1()	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)	Unknown
 	WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeImpl()	Unknown
 	WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(object state)	Unknown
 	WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(object obj)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	WindowsBase.dll!MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	WindowsBase.dll!System.Windows.Threading.DispatcherOperation.Invoke()	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.ProcessQueue()	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.WndProcHook(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)	Unknown
 	WindowsBase.dll!MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)	Unknown
 	WindowsBase.dll!MS.Win32.HwndSubclass.DispatcherCallbackOperation(object o)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority priority, System.TimeSpan timeout, System.Delegate method, object args, int numArgs)	Unknown
 	WindowsBase.dll!MS.Win32.HwndSubclass.SubclassWndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam)	Unknown
```

**Comment by gundermanc** [comment]
Taking a look from the IDE side. This looks like a deadlock..

**Comment by huguesv** [comment]
@starship863 we've determined that this is fixed in the upcoming 15.7 Preview 4.
